### PR TITLE
add terminology lookup operator

### PR DESCRIPTION
**Pull request by jdroenner**
blocked by https://github.com/umr-dbs/mapping-gfbio/pull/3

### CODE REVIEW

File: src/app/operators/dialogs/terminology-lookup/terminology-lookup.component.ts [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {Observable} from 'rxjs/Rx';
+import {Operator} from '../../operator.model';
+import {ProjectService} from '../../../project/project.service';
+import {WaveValidators} from '../../../util/form.validators';
+
+import {RandomColorService} from '../../../util/services/random-color.service';
+import {TerminologyLookupOnNotResolvable, TerminologyLookupType} from '../../types/terminology-lookup-type';
+
+// TODO: replace with actual lookup
```
[the remark this answers is not in the document]
Comment (jdroenner) [reply src/app/operators/dialogs/terminology-lookup/terminology-lookup.component.ts]: will do this when we the rest works

File: src/app/operators/dialogs/terminology-lookup/terminology-lookup.component.ts [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+                terminology_key: terminologyKeyName,
+                on_not_resolvable: on_not_resolvable,
+                define_new_column_appendix: define_new_column_appendix,
+                new_column_appendix: new_column_appendix,
+            }),
+            resultType: sourceOperator.resultType,
+            projection: sourceOperator.projection,
+            attributes: operatorAttributes,
+            dataTypes: sourceOperator.dataTypes,
+            units: sourceOperator.units,
```
[the remark this answers is not in the document]
Comment (jdroenner) [reply src/app/operators/dialogs/terminology-lookup/terminology-lookup.component.ts]: is everything from a terminology a string?

[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply src/app/operators/dialogs/terminology-lookup/terminology-lookup.component.ts]: I think so.

**Comment by jdroenner** [comment]
is this still blocked?

**Pull request closed without merging**